tools/validate/validate.cpp

```diff
@@ -19,15 +19,16 @@ constexpr size_t max_line_length = 120;
 
 class BinaryFile {
 public:
-    explicit BinaryFile(const filesystem::path& filepath) {
-        m_file = _wfopen(filepath.c_str(), L"rb");
-
+    explicit BinaryFile(const filesystem::path& filepath)
+        // clang-format off
+    : m_file(_wfopen(filepath.c_str(), L"rb")) {
+        // clang-format on
         if (!m_file) {
             fwprintf(stderr, L"Validation failed: %ls couldn't be opened.\n", filepath.c_str());
         }
     }
 
-    [[nodiscard]] bool read_next_block(vector<unsigned char>& buffer) {
+    [[nodiscard]] bool read_next_block(vector<unsigned char>& buffer) const {

```

Comment:
No change requested: The addition of `const` here is debatable, but I have no objection to this.

Reply:
Ehh, I'm not the biggest fan, since you're modifying the underlying `FILE*`'s state. It's true that it's implemented as a pointer, so it's still technically `const`; however, imo, `const` member functions should have the same behavior if called multiple times on the same object.

Reply:
Ok, cool. Pushed changes to restore how I originally implemented this.